packages/typespec-ts/test/integration/routes.spec.ts

```diff
@@ -32,13 +32,94 @@ describe("RoutesClient Rest Client", () => {
     assert.strictEqual(result.status, "204");
   });
 
-  it.skip("should have PathParameters SimpleExpansion Standard primitive", async () => {
+  it("should have allowReserved: true", async () => {
     const result = await client
-      .path("/routes/path/simple/standard/primitive{param}", "a")
+      .path("/routes/path/reserved-expansion/template/{param}", {
+        value: "foo/bar baz",
+        allowReserved: true
+      })
       .get();
     assert.strictEqual(result.status, "204");
   });
 
+  it("should have explode: true array", async () => {
+    const result = await client
+      .path("/routes/query/query-expansion/explode/array")
+      .get({
+        queryParameters: {
+          param: {
+            value: ["a", "b"],
+            explode: true,
+            style: "form"
+          }
+        }
+      });
+    assert.strictEqual(result.status, "204");
+  });
+
+  it("should have explode: true record", async () => {
+    const result = await client
+      .path("/routes/query/query-expansion/explode/record")
+      .get({
+        queryParameters: {
+          param: {
+            value: { a: 1, b: 2 },
+            explode: true,
+            style: "form"
+          }
+        }
+      });
+    assert.strictEqual(result.status, "204");
+  });
+
+  it("should have explode: true primitive", async () => {
+    const result = await client
+      .path("/routes/query/query-expansion/explode/primitive")
+      .get({
+        queryParameters: {
+          param: {
+            value: "a",
+            explode: true,
+            style: "form"
+          }
+        }
+      });
+    assert.strictEqual(result.status, "204");
+  });
+
+  it("should have explode: false array", async () => {
+    const result = await client
+      .path("/routes/query/query-expansion/standard/array")
+      .get({
+        queryParameters: {
+          param: ["a", "b"]
+        }
+      });
+    assert.strictEqual(result.status, "204");
+  });
+
+  it("should have explode: false record", async () => {
```

Comment:
I re-thinked this a little and it would be problematic if we convert any object to format like a,1,b,2 in core e.g if this is a Date, toString would be date string but we may not know how to format as form style. maybe here we should always generate it explictly in codegen.
As interface Param { value: Record<string, string>; explode: false; style: "form" }.

Reply:
Honestly it feels strange to me that this is considered a valid case. I don't really understand why we would serialize an object in this way. Is there a reference (specification or something) that describes this particular behavior?

Reply:
Never mind, I see it described here: https://swagger.io/docs/specification/v3_0/describing-parameters/. We can implement that, but like you say there might be a problem for objects where `toString` actually provides a meaningful value like Date. Do you know if there are any existing cases where RLC generates a type like that for a query parameter?

Reply:
How about, if the object has a custom `toString` implementation, use that, otherwise expand it?